### ISSUE TITLE
Add dependabot to the server repo

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/server#9592]
We have quite some dependencies besides the 3rdparty repo, that we should keep track of if we use the latest versions. Also with the newest vue based settings/apps it would make sense to habe some automatic check for outdated dependencies. I'd suggest to enable dependabot (https://app.dependabot.com) for the server repo to check the following files for now:
- build/integration/composer.json
- build/package.json
- settings/packages.json
- apps/updatenotification/package.json

This would also help us to identify issues like #9407 and https://github.com/nextcloud/server/pull/9348 earlier.

cc @skjnldsv @MorrisJobke @ChristophWurst

**Comment by juliushaertl** [comment]
@nextcloud/javascript

**Comment by ChristophWurst** [comment]
Yes, please. This would make a lot of sense to enable and dependabot is capable of handling these package manager file in subdirectories. The only limitation right now is that we cannot use vendoring for the 3rd-party repo due to https://github.com/dependabot/dependabot-core/issues/386.

I'd suggest to start with one of the files and see how that work and then enable if for the rest step-by-step (I expect loads of PRs).
